rider-fsharp/src/main/java/com/jetbrains/rider/ideaInterop/fileTypes/fsharp/lexer/FSharpTokenType.java

```diff
@@ -0,0 +1,255 @@
+    IElementType EXTERN = createKeywordToken("EXTERN", "extern");
+    IElementType FALSE = createKeywordToken("FALSE", "false");
+    IElementType FINALLY = createKeywordToken("FINALLY", "finally");
+    IElementType FOR = createKeywordToken("FOR", "for");
+    IElementType FUN = createKeywordToken("FUN", "fun");
+    IElementType FUNCTION = createKeywordToken("FUNCTION", "function");
+    IElementType GLOBAL = createKeywordToken("GLOBAL", "global");
+    IElementType IF = createKeywordToken("IF", "if");
+    IElementType IN = createKeywordToken("IN", "in");
+    IElementType INHERIT = createKeywordToken("INHERIT", "inherit");
+    IElementType INLINE = createKeywordToken("INLINE", "inline");
+    IElementType INTERFACE = createKeywordToken("INTERFACE", "interface");
+    IElementType INTERNAL = createKeywordToken("INTERNAL", "internal");
+    IElementType LAZY = createKeywordToken("LAZY", "lazy");
+    IElementType LET = createKeywordToken("LET", "let");
+    IElementType MATCH = createKeywordToken("MATCH", "match");
+    IElementType MEMBER = createKeywordToken("MEMBER", "member");
+    IElementType MODULE = createKeywordToken("MODULE", "module");
+    IElementType MUTABLE = createKeywordToken("MUTABLE", "mutable");
+    IElementType NAMESPACE = createKeywordToken("NAMESPACE", "namespace");
+    IElementType NEW = createKeywordToken("NEW", "new");
+    IElementType NULL = createKeywordToken("NULL", "null");
+    IElementType OF = createKeywordToken("OF", "of");
+    IElementType OPEN = createKeywordToken("OPEN", "open");
+    IElementType OR = createKeywordToken("OR", "or");
+    IElementType OVERRIDE = createKeywordToken("OVERRIDE", "override");
+    IElementType PRIVATE = createKeywordToken("PRIVATE", "private");
+    IElementType PUBLIC = createKeywordToken("PUBLIC", "public");
+    IElementType REC = createKeywordToken("REC", "rec");
+    IElementType RETURN = createKeywordToken("RETURN", "return");
+    IElementType SIG = createKeywordToken("SIG", "sig");
+    IElementType STATIC = createKeywordToken("STATIC", "static");
+    IElementType STRUCT = createKeywordToken("STRUCT", "struct");
+    IElementType THEN = createKeywordToken("THEN", "then");
+    IElementType TO = createKeywordToken("TO", "to");
+    IElementType TRUE = createKeywordToken("TRUE", "true");
+    IElementType TRY = createKeywordToken("TRY", "try");
+    IElementType TYPE = createKeywordToken("TYPE", "type");
+    IElementType UPCAST = createKeywordToken("UPCAST", "upcast");
+    IElementType USE = createKeywordToken("USE", "use");
+    IElementType VAL = createKeywordToken("VAL", "val");
+    IElementType VOID = createKeywordToken("VOID", "void");
+    IElementType WHEN = createKeywordToken("WHEN", "when");
+    IElementType WHILE = createKeywordToken("WHILE", "while");
+    IElementType WITH = createKeywordToken("WITH", "with");
+    IElementType YIELD = createKeywordToken("YIELD", "yield");
+
+    TokenSet OP_KEYWORD = TokenSet.create(
+            LET_BANG,
```

Comment:
Is there any reason to have `do!`, `match!` and other keywords in a separate set?